wallet/internal/db/safecasting_test.go

```diff
@@ -0,0 +1,236 @@
+package db
+
+import (
+	"database/sql"
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestInt64ToUint32 checks that an int64 value is converted to uint32 only
+// when it is non-negative and fits within the uint32 range. It should fail
+// loudly for any value outside those bounds.
+func TestInt64ToUint32(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name    string
+		val     int64
+		want    uint32
+		wantErr bool
```

Comment:
sth for future dev: we can define `errExpected error` or `errWanted error`, then pass nil for cases where errors are not expected, in the assertion we can use a single line `require.ErrorIs` without the logical branch.